plugins/modules/system/pamd.py

```diff
@@ -750,22 +748,25 @@ def __str__(self):
         return '\n'.join(lines) + '\n'
 
 
-def parse_module_arguments(module_arguments):
-    # Return empty list if we have no args to parse
-    if not module_arguments:
-        return []
-    elif isinstance(module_arguments, list) and len(module_arguments) == 1 and not module_arguments[0]:
-        return []
+def parse_module_arguments(module_arguments, return_none=False):
+    # If args is None, return empty list by default.
+    # But if return_none is True, then return None
+    if module_arguments is None:
+        return None if return_none else []
+    if isinstance(module_arguments, list) and len(module_arguments) == 1 and module_arguments[0] is None:
```

Comment:
I think this needs to stay as it was before, since if it's a list in this case, it's most likely `['']` and not `[None]`:
```suggestion
    if isinstance(module_arguments, list) and len(module_arguments) == 1 and not module_arguments[0]:
```